tracing-appender/src/rolling.rs

```diff
@@ -93,19 +93,11 @@ pub fn never(directory: impl AsRef<Path>, file_name_prefix: impl AsRef<Path>) ->
     )
 }
 
-pub(crate) trait WriterFactory: Debug + Clone + Send {
-    type W: Write + Debug + Send;
-
-    fn create_writer(&self, directory: &str, filename: &str) -> io::Result<Self::W>;
-}
-
 #[derive(Clone, Debug)]
 pub(crate) struct BufWriterFactory {}
 
-impl WriterFactory for BufWriterFactory {
-    type W = BufWriter<File>;
-
-    fn create_writer(&self, directory: &str, filename: &str) -> io::Result<BufWriter<File>> {
+impl BufWriterFactory {
+    pub fn create_writer(&self, directory: &str, filename: &str) -> io::Result<BufWriter<File>> {
```

Comment:
If we're no longer generic over a writer factory, does this need to exist as a separate type still? Can we just move this code to `InnerAppender`? Having a separate type seems needlessly complicated.

Reply:
Sure, will update this now